dspace-api/src/main/java/org/dspace/core/HibernateDBConnection.java

```diff
@@ -164,6 +164,83 @@ private void configureDatabaseMode() throws SQLException {
      */
     @Override
     public <E extends ReloadableEntity> void uncacheEntity(E entity) throws SQLException {
-        getSession().evict(entity);        
+        if(entity != null) {
+            if (entity instanceof DSpaceObject) {
+                DSpaceObject dso = (DSpaceObject) entity;
+
+                // The metadatavalue relation has CascadeType.ALL, so they are evicted automatically
+                // and we don' need to uncache the values explicitly.
+
+                if(Hibernate.isInitialized(dso.getHandles())) {
```

Comment:
No it won't lead to problems. Hibernate will only touch the handles if they have been retrieved from the database before. The `Utils.emptyIfNull()` will make sure we don't get any null pointer exceptions in case the handles collection is actually null.